packages/flutter_tools/lib/src/commands/channel.dart

```diff
@@ -24,18 +34,20 @@ class ChannelCommand extends FlutterCommand {
   Future<Null> runCommand() {
     switch (argResults.rest.length) {
       case 0:
-        return _listChannels();
+        return listChannels(showAll: argResults['all']);
       case 1:
-        return _switchChannel(argResults.rest[0]);
+        return switchChannel(argResults.rest[0]);
       default:
         throw new ToolExit('Too many arguments.\n$usage');
     }
   }
 
-  Future<Null> _listChannels() async {
-    final String currentBranch = runSync(
-        <String>['git', 'rev-parse', '--abbrev-ref', 'HEAD'],
-        workingDirectory: Cache.flutterRoot);
+  static Future<Null> listChannels({ bool showAll }) async {
+    // Beware: currentBranch could contain PII. See getBranchName().
```

Comment:
But `getBranchName()` redacts it if it's not a known branch name, so we should be safe here

Reply:
Ah, scratch that - we don't set `whitelistBranchName` to true here